src/components/FilterBar/FilterDisplay/Inputs/DateInput.react.js

```diff
@@ -24,7 +24,11 @@ export class DateInput extends React.Component {
   }
 
   onBlur() {
-    this.context.filterBarActor.updateFilter(this.props.filterUid, "value", this.state.value);
+    if (this.props.onBlur) {
+      this.props.onBlur();
+    } else {
+      this.context.filterBarActor.updateFilter(this.props.groupKey, this.props.inputKey, this.state.value);
+    }
```

Comment:
I didn't quite follow this change. Why filter types are being affected by groups? Is there any reason where a Filter needs to know the group it belongs to?

Reply:
@msxavi yes it's a bit tricky and most complicated part of the code.
As I mentioned in PR descripute now component has new data strucure it two-sized array `activeFilters` in `FilterBarStore`.
It can be presented like this:
```js
this.activeFilter = [
  [Filter1, Filter2], // <- Group1
  [Filter3], // <- Group2
  [Filter4, Filter5] // <- Group3
]
```

As you can see the external array presents groups and interal array presents filters inside each group.
Groups/Filters rendering based on  that `activeFitlers` variable and to identify correct pathway to manage data-values we heed `groupKey` which represents group index in external array and we need `inputKey` which represents filter index in internal array. When we change values in filter component we should update data in `Store` and set updated values in `activeFilters` and specially in this case we should use `groupKey` and `inputKey` like this:
```js
this.activeFilters[groupKey][inputKey] = newFilterValue;
```